### PR TITLE
Adds AnthropicChat LLM class for interfacing with Claude

### PR DESCRIPTION
**Pull request by jacoblee93**
This PR adds support for Anthropic's recently released chatbot, Claude, as a LangChain LLM by creating a wrapper `AnthropicChat` class around Anthropic's TypeScript SDK.

I based the new class of the existing `OpenAIChat` class, including adding support for users to pass in OpenAI style `prefixMessages` (`ChatCompletionRequestMessage` from their SDK). The new class will translate those messages into Anthropic's very particular prompting syntax, allowing for a relatively similar user-facing API.

This PR adds a dependency on Anthropic's official `@anthropic-ai/sdk`, and requires an Anthropic API key set as an environment variable named `ANTHROPIC_API_KEY` environment variable for the new test cases to pass, so it may require an update to the CI/CD runner.

### CODE REVIEW

**Comment by conroywhitney** [comment]
I don't have Anthropic yet to test this out 😢 but I'm looking forward to speaking with Claude!

Since you added a new `.env` var, would you mind updating `langchain/.env.example` and `examples/.env.example`?

I noticed the two actually differed slightly, so here's a combined list (including the new `ANTHROPIC_API_KEY`):

```
ANTHROPIC_API_KEY=ADD_YOURS_HERE
COHERE_API_KEY=ADD_YOURS_HERE
HUGGINGFACEHUB_API_KEY=ADD_YOURS_HERE
OPENAI_API_KEY=ADD_YOURS_HERE
PINECONE_API_KEY=ADD_YOURS_HERE
PINECONE_ENVIRONMENT=ADD_YOURS_HERE
PINECONE_INDEX=ADD_YOURS_HERE
SERPAPI_API_KEY=ADD_YOURS_HERE
SERPER_API_KEY=ADD_YOURS_HERE
SUPABASE_PRIVATE_KEY=ADD_YOURS_HERE
SUPABASE_URL=ADD_YOURS_HERE
```

**Comment by nfcampos** [comment]
Thanks for this. I think ideally this would be a subclass of ChatModel, not LLM

**Comment by nfcampos** [comment]
@jacoblee93 thanks! You can ignore that combine method, it is optional, and as you say doesn't apply to this one

File: langchain/src/chat_models/anthropic.ts [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+          },
+        });
+      };
+    } else {
+      if (!this.batchClient) {
+        this.batchClient = new AnthropicApi(this.apiKey);
+      }
+      makeCompletionRequest = async () => this.batchClient.complete(request);
+    }
+    return backOff(makeCompletionRequest, {
```
Comment (nfcampos) [inline langchain/src/chat_models/anthropic.ts:288]: Replace this with `AsyncCaller`

Comment (jacoblee93) [reply langchain/src/chat_models/anthropic.ts]: Swapped it out!

**Comment by jacoblee93** [comment]
PTAL @nfcampos @agola11

**Comment by jacoblee93** [comment]
Ah whoops, I merged my personal `main` branch and not the upstream one - will fix now.

**Comment by jacoblee93** [comment]
Alright, merged the correct `main` branch this time!

File: langchain/src/chat_models/tests/chatanthropic.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const responseA = await chat.generatePrompt([
+    await chatPrompt.formatPromptValue({
+      input_language: "English",
+      output_language: "French",
+      text: "I love programming.",
+    }),
+  ]);
+
+  console.log(responseA.generations);
+}, 50000);
```
Comment (agola11) [inline langchain/src/chat_models/tests/chatanthropic.int.test.ts:106]: This is not needed, as I believe @nfcampos added a limit for all int tests

**Review by agola11 — APPROVED** [review]
This lgtm, just fix CI

File: langchain/src/chat_models/tests/chatanthropic.int.test.ts (hunk 2 above)
Comment (jacoblee93) [reply langchain/src/chat_models/tests/chatanthropic.int.test.ts]: Ok, just removed the timeouts.

**Comment by jacoblee93** [comment]
I think the last failing test is some approvals check? This shouldn't touch any docs pages.

**Pull request merged**